vault/resource_aws_auth_backend_roletag_blacklist.go

```diff
@@ -0,0 +1,153 @@
+package vault
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/vault/api"
+)
+
+var (
+	awsAuthBackendRoleTagBlacklistBackendFromPathRegex = regexp.MustCompile("^auth/(.+)/config/tidy/roletag-blacklist$")
+)
+
+func awsAuthBackendRoleTagBlacklistResource() *schema.Resource {
+	return &schema.Resource{
+		Create: awsAuthBackendRoleTagBlacklistWrite,
+		Read:   awsAuthBackendRoleTagBlacklistRead,
+		Update: awsAuthBackendRoleTagBlacklistWrite,
+		Delete: awsAuthBackendRoleTagBlacklistDelete,
+		Exists: awsAuthBackendRoleTagBlacklistExists,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"backend": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Unique name of the auth backend to configure.",
+				ForceNew:    true,
+				Default:     "aws",
+				// standardise on no beginning or trailing slashes
+				StateFunc: func(v interface{}) string {
+					return strings.Trim(v.(string), "/")
+				},
+			},
+			"safety_buffer": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Description: "The amount of extra time that must have passed beyond the roletag expiration, before it's removed from backend storage.",
+			},
+			"disable_periodic_tidy": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "If true, disables the periodic tidying of the roletag blacklist entries.",
+			},
+		},
+	}
+}
+
+func awsAuthBackendRoleTagBlacklistWrite(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*api.Client)
+
+	backend := d.Get("backend").(string)
+	data := map[string]interface{}{}
+
+	if v, ok := d.GetOk("safety_buffer"); ok {
+		data["safety_buffer"] = v
+	}
+	if v, ok := d.GetOk("disable_periodic_tidy"); ok {
```

Comment:
`disable_periodic_tidy` needs a default value specified, or we need to use `GetOkExists` here otherwise `disable_periodic_tidy` can never be set to `false`, and we'll never enter this block